google/resource_cloudfunctions_function_test.go

```diff
@@ -0,0 +1,475 @@
+  source = "%s"
+}
+
+resource "google_cloudfunctions_function" "function" {
+  name                  = "%s"
+  description           = "test function"
+  available_memory_mb   = 128
+  source_archive_bucket = "${google_storage_bucket.bucket.name}"
+  source_archive_object = "${google_storage_bucket_object.archive.name}"
+  trigger_http          = true
+  timeout		        = 61
+  entry_point           = "helloGET"
+  labels {
+	my-label = "my-label-value"
+  }
+}
+`, bucketName, zipFilePath, functionName)
+}
+
+func testAccCloudFunctionsFunction_updated(functionName string, bucketName string, zipFilePath string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name = "%s"
+}
+
+resource "google_storage_bucket_object" "archive" {
+  name   = "index.zip"
+  bucket = "${google_storage_bucket.bucket.name}"
+  source = "%s"
+}
+
+resource "google_cloudfunctions_function" "function" {
+  name                  = "%s"
+  description           = "test function updated"
+  available_memory_mb   = 256
+  source_archive_bucket = "${google_storage_bucket.bucket.name}"
+  source_archive_object = "${google_storage_bucket_object.archive.name}"
+  trigger_http          = true
+  timeout		        = 91
+  entry_point           = "helloGET"
+  labels {
+	my-label = "my-updated-label-value"
+	a-new-label = "a-new-label-value"
+  }
+}`, bucketName, zipFilePath, functionName)
+}
+
+func testAccCloudFunctionsFunction_pubsub(functionName string, bucketName string,
+	topic string, zipFilePath string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name = "%s"
+}
+
+resource "google_storage_bucket_object" "archive" {
+  name   = "index.zip"
+  bucket = "${google_storage_bucket.bucket.name}"
+  source = "%s"
+}
+
+resource "google_pubsub_topic" "sub" {
+  name = "%s"
+}
+
+resource "google_cloudfunctions_function" "function" {
+  name                  = "%s"
+  available_memory_mb   = 128
+  source_archive_bucket = "${google_storage_bucket.bucket.name}"
+  source_archive_object = "${google_storage_bucket_object.archive.name}"
+  trigger_topic         = "${google_pubsub_topic.sub.name}"
+  timeout		        = 61
+  entry_point           = "helloPubSub"
+}`, bucketName, zipFilePath, topic, functionName)
+}
+
+func testAccCloudFunctionsFunction_bucket(functionName string, bucketName string,
+	zipFilePath string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name = "%s"
+}
+
+resource "google_storage_bucket_object" "archive" {
+  name   = "index.zip"
+  bucket = "${google_storage_bucket.bucket.name}"
+  source = "%s"
+}
+
+resource "google_cloudfunctions_function" "function" {
+  name                  = "%s"
+  available_memory_mb   = 128
+  source_archive_bucket = "${google_storage_bucket.bucket.name}"
+  source_archive_object = "${google_storage_bucket_object.archive.name}"
+  trigger_bucket        = "${google_storage_bucket.bucket.name}"
+  timeout		        = 61
```

Comment:
and here

Reply:
Fixed, sorry for this - my editor didn't show those tabs...